Implement time to live (#7327)
nextcloud/spreed: docs/conversation.md

        `participantFlags` | int | v1 | | "In call" flags of the user's session making the request (only available with `in-call-flags` capability)
        `readOnly` | int | v1 | | Read-only state for the current user (only available with `read-only-rooms` capability)
        `listable` | int | v3 | | Listable scope for the room (only available with `listable-rooms` capability)

>         `expireDate` | int | v4 | | The expire date time in seconds in this chat. Zero if disabled. (only available with `message-expire` capability)

Author: We already approved as expireDate in server. Do you preffer to change all, including the server methods, to expireMessage?
https://github.com/nextcloud/server/blob/master/lib/public/Comments/IComment.php#L310

>         `expireDate` | int | v4 | | The expire date time in seconds in this chat. Zero if disabled. (only available with `message-expire` capability)

Reviewer: Server methods are fine as they are on the Comments object and table so there it is clear that it expires the comment only.

        `statusMessage` | string | v4 | | Optional: Only available for one-to-one conversations and when  `includeStatus=true` is set
        `participants` | array | v1 | v2 | **Removed**
        `guestList` | string | v1 | v2 | **Removed**

>         `timeToLive` | int | v1 | The time to live of messages in this chat. Zero if disabled.

Reviewer: ```suggestion
        `timeToLive` | int | v4 | | The time to live of messages in this chat. Zero if disabled. (only available with `time-to-live` capability)
```

>         `timeToLive` | int | v4 | The time to live of messages in this chat. Zero if disabled. (only available with `time-to-live` capability)

Reviewer: ```suggestion
        `timeToLive` | int | v4 | | The time to live of messages in this chat. Zero if disabled. (only available with `time-to-live` capability)
```
Empty column is needed, so the Description is in the right column:

<img width="975" alt="2022-05-09 19_50_44-Implement time to live by vitormattos · Pull Request #7327 · nextcloud_spreed" src="https://user-images.githubusercontent.com/213943/167468253-9aa64fe1-e1f5-4da9-84e9-2de42f4fbc16.png">

Also it should be above `participants` so removed items are at the end.

        + `401 Unauthorized` When the participant is a guest
        + `404 Not Found` When the conversation could not be found for the participant

> ## Set time to live for messages in a conversation

Reviewer: Jan doesn't like "ttl" and "time to live"
Maybe we rename the database field to "chat_expiration"
and also rename the value on the APIs and the capabilities?

* Required capability: `time-to-live`
* Method: `POST`
* Endpoint: `/room/{token}/ttl`
* Data:

    field | type | Description
    ---|---|---

>     `ttl` | int | The time to live for messages in seconds. If is zero, will disable the ttl.

Reviewer: ```suggestion
    `ttl` | int | The time to live for messages in seconds. If is zero, messages will not be deleted automatically.
```

        + `401 Unauthorized` When the participant is a guest
        + `404 Not Found` When the conversation could not be found for the participant

> ## Set time to live of messages for a conversation

Reviewer: ```suggestion
## Set time to live for messages in a conversation
```

* Required capability: `time-to-live`
* Method: `POST`
* Endpoint: `/room/{token}/ttl`
* Data:

    field | type | Description
    ---|---|---
    `ttl` | int | The time to live of messages in seconds. Must be greater than zero.

* Response:
    - Status code:
        + `200 OK`
        + `400 Bad Request` Invalid value
        + `403 Forbidden` When the current user is not a moderator, owner or guest moderator
        + `404 Not Found` When the conversation could not be found for the participant

## Disable the time to live of messages for a conversation

* Required capability: `time-to-live`
* Method: `DELETE`
* Endpoint: `/room/{token}/ttl`

* Response:
    - Status code:
        + `200 OK`
        + `403 Forbidden` When the current user is not a moderator, owner or guest moderator

>         + `404 Not Found` When the conversation could not be found for the participant

Reviewer: Since 0 means disabled, we can also just do a post with 0? Less endpoints less breaking?
